sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/owaspzapwrapper/helper/SecHubIncludeExcludeToOwaspZapURIHelperTest.java

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.owaspzapwrapper.helper;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SecHubIncludeExcludeToOwaspZapURIHelperTest {
+
+    private SecHubIncludeExcludeToOwaspZapURIHelper helperToTest;
+
+    @BeforeEach
+    void beforeEach() {
+        helperToTest = new SecHubIncludeExcludeToOwaspZapURIHelper();
+    }
+
+    @Test
+    void returns_empty_list_if_list_of_subSites_is_null() {
+        /* prepare */
+        String targetUrl = "https://127.0.0.1:8080";
+        List<String> sites = null;
+
+        /* execute */
+        List<String> urls = helperToTest.createListOfUrls(targetUrl, sites);
+
+        /* test */
+        assertTrue(urls.isEmpty());
+    }
+
+    @Test
+    void returns_empty_list_if_list_of_subSites_is_empty() {
+        /* prepare */
+        String targetUrl = "https://127.0.0.1:8080";
+        List<String> sites = new ArrayList<>();
+
+        /* execute */
+        List<String> urls = helperToTest.createListOfUrls(targetUrl, sites);
+
+        /* test */
+        assertTrue(urls.isEmpty());
+    }
+
+    @Test
+    void returns_list_of_url_conform_for_owasp_zap_includes_or_excludes() {
+        /* prepare */
+        String targetUrl = "https://127.0.0.1:8080";
+        List<String> sites = createExampleListOfSites();
+
+        /* execute */
+        List<String> urls = helperToTest.createListOfUrls(targetUrl, sites);
+
+        /* test */
+        assertTrue(urls.contains("https://127.0.0.1:8080/sub.*"));
+        assertTrue(urls.contains("https://127.0.0.1:8080/sub1/directory.*"));
+        assertTrue(urls.contains("https://127.0.0.1:8080/sub2/directory.*"));
+        assertTrue(urls.contains("https://127.0.0.1:8080/sub3/directory.*"));
+        assertTrue(urls.contains("https://127.0.0.1:8080/sub4/directory.*"));
+    }
+
+    private List<String> createExampleListOfSites() {
+        List<String> subSites = new ArrayList<>();
+        subSites.add("/sub/");
```

Comment:
What happens if I want to include or exclude a single URL? For example:`https://127.0.0.1:8080/xyz/index.html`. Does that work?

Reply:
Single directories cannot be excluded right now, because the regular expression `.*` is appended to every URI. The URI from your example would be changed to `https://127.0.0.1:8080/xyz/index.html.*`, which works since it is a file. If we want to exclude single directories, I have to think of a different solution.